File: front-end/src/api/video.ts
import request from '../util/request'
import {
    Video
} from '@/types/video.interface'
import {
    PageQuery
} from '@/types/pagequery.interface'
import {
    MeasObjectVo
} from '@/types/measobject.interface'

export function getDefaultVideo(): Video {
    return {
        id: 0,
        name: '',
        stationId: 0,
        stationName: '',
        objectType: 0,
        objectTypeName: '',
        dataType: 0,
        dataTypeName: '',
        equipmentSn: '',
        createTime: new Date(),
        ip: '127.0.0.1',
        port: 80,
        username: 'admin',
        password: '12345',
        serverId: 0,
        channelNo: 0,
        channelName: '',
        defaultPreset: 0,
        vendor: 0,
        videoServer: null
    }
}

/** 获取所有的监测视频 */
export function getVideos() {
    return request({
        url: 'videos',
        method: 'get'
    })
}

/** 分页获取视频 */
export function getPageInfo(param: PageQuery) {
    return request({
        url: 'videos_page',
        method: 'post',
        data: param
    })
}

/** 添加视频 */
export function addVideo(video: Video) {
    return request({
        url: 'videos',
        method: 'post',
        data: video
    })
}

/** 编辑视频 */
export function editVideo(video: Video) {
    return request({
        url: 'videos_edit',
        method: 'post',
        data: video
    })
}

/** 删除视频 */
export function deleteVideo(id: number) {
    return request({
        url: 'videos_delete/' + id,
        method: 'get'
    })
}

export function getVideoDtos() {
    return request({
        url: 'videodtos',
        method: 'get'
    })
}

export function getVideoDto(id: number) {
    return request({
        url: 'videodtos/' + id,
        method: "get"
    })
}

/** 获取场站下的所有视频 */
export function getVideoDtosByStation(stationId: number) {
    return request({
        url: `stations/${stationId}/videodtos`,
        method: 'get'
    })
}

/** 根据条件获得视频 */
export function getVideoDtosByCondition(vo: MeasObjectVo) {
    return request({
        url: 'videodtos/condition',
        method: 'post',
        data: vo
    })
}

File: back-end/src/main/java/com/bandweaver/maxsec/entity/Video.java
package com.bandweaver.maxsec.entity;

import com.alibaba.fastjson.JSONObject;
import lombok.Data;

/**
 * 视频
 */
@Data
public class Video extends MeasObject {

    /**
     * ip
     */
    private String ip;

    /**
     * 端口号
     */
    private Integer port;

    /**
     * 用户名
     */
    private String username;

    /**
     * 密码
     */
    private String password;

    /**
     * 所属服务
     */
    private Integer serverId;

    /**
     * 通道号
     */
    private Integer channelNo;

    /**
     * 通道名称
     */
    private String channelName;

    /**
     * 默认预置位
     */
    private Integer defaultPreset;

    /**
     * 类型
     */
    private Integer vendor;

    public static Video fromMeasObj(MeasObject obj) {
        Video tmp = new Video();
        String str = JSONObject.toJSONString(obj);
        tmp = (Video) JSONObject.parseObject(str, Video.class);
        return tmp;
    }

    public String toJson() {
        return JSONObject.toJSONString(this);
    }

    public static Video fromJson(String json) {
        return (Video) JSONObject.parseObject(json, Video.class);
    }
}
